integrationtest/mapper/orm-spring-uberjar/test/src/test/java/org/hibernate/search/integrationtest/spring/repackaged/test/RepackagedAppIT.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.spring.repackaged.test;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Path;
+import java.util.jar.JarEntry;
+
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.ProjectionConstructor;
+import org.hibernate.search.util.common.jar.impl.JandexUtils;
+
+import org.junit.Test;
+
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.FieldInfo;
+import org.jboss.jandex.Index;
+
+import acme.org.hibernate.search.integrationtest.spring.repackaged.model.MyEntity;
+import acme.org.hibernate.search.integrationtest.spring.repackaged.model.MyProjection;
+import org.springframework.boot.loader.jar.JarFile;
+
+public class RepackagedAppIT {
```

Comment:
Actually scratch that, since the end-to-end integration test just executes an app, so it might not detect regressions such as "we open the nested JAR but it's empty"